src/bindings/python/utils.py

```diff
@@ -0,0 +1,58 @@
+def decode_polyline(polyline, precision=6, order='lnglat'):
+    """Decodes an encoded polyline string to a list of coordinates.
+
+    :param str polyline: An encoded polyline string.
+    :param int precision: The coordinate precision, default 6.
+    :param str order: Which order the coordinates are stored in, 'latlng' or 'lnglat'.
+
+    :returns: List of decoded coordinates in specified order.
+    :rtype: List[Tuple[float, float]]
+    """
+
+    return _decode(polyline, precision=precision, order=order, is3d=False)
+
+
+def _trans(value, index):
+    """
+    Copyright (c) 2014 Bruno M. Custódio
+    Copyright (c) 2016 Frederick Jansen
+    https://github.com/hicsail/polyline/commit/ddd12e85c53d394404952754e39c91f63a808656
+    """
+    byte, result, shift = None, 0, 0
+
+    while byte is None or byte >= 0x20:
+        byte = ord(value[index]) - 63
+        index += 1
+        result |= (byte & 0x1f) << shift
+        shift += 5
+        comp = result & 1
+
+    return ~(result >> 1) if comp else (result >> 1), index
+
+
+def _decode(expression, precision=5, order='lnglat', is3d=False):
```

Comment:
we already have sample code for this in python in the docs in this repo, should probably go with that unless its not working for you: https://github.com/valhalla/valhalla/blob/master/docs/decoding.md#python

Reply:
or better yet just expose the c++ functions in the python bindings and not have to maintain two functions for doing this :smile:

Reply:
right, I'd take the latter :) I'll remove it for now and save it for a future PR.